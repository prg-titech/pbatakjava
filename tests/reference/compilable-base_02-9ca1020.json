{
    "basename": "compilable-base_02-9ca1020",
    "cmd": "{infile}",
    "infile": "tests/base_02/build.sh",
    "infile_hash": "711f091c317e8d3f96c224710cce66e4299b96929c9d66871c9cbd1b",
    "stdout": "compilable-base_02-9ca1020.stdout",
    "stdout_hash": "351fd97afeabd3ced7c28b3e3b7d6685e0c35b4eadde2f10d25dd94b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}