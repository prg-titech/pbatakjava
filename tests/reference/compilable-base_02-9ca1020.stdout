
public interface Point {
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  private int x;

  private int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public interface Line {
  public Point get__p1();

  public Point set__p1(Point toAssign);

  public Point get__p2();

  public Point set__p2(Point toAssign);
}

public interface Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2);
}

public class Line_v1 implements Line {
  public Point_Factory V;

  public Point_Factory W;

  public Point p1;

  public Point p2;

  public Line_v1(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    this.V = V;
    this.W = W;
    this.p1 = p1;
    this.p2 = p2;
  }

  public Point get__p1() {
    return this.p1;
  }

  public Point set__p1(Point toAssign) {
    return this.p1 = toAssign;
  }

  public Point get__p2() {
    return this.p2;
  }

  public Point set__p2(Point toAssign) {
    return this.p2 = toAssign;
  }
}

public class Line_v1_Factory implements Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    return new Line_v1(V, W, p1, p2);
  }
}
